TaskHero/Step.swift

```diff
@@ -13,5 +13,66 @@ class Step: NSObject {
     var details: String?
     var state: String?
     var assignees: [User]?
-    var signoffs: [User]?
+    var signoff: User?
+    var completedBy: User?
+    var completedAt: TimeInterval?
+    
+    override init() {
+        super.init()
+    }
+    
+    init(stepDictionary: [String: AnyObject]?) {
+        super.init()
+        
+        self.name = stepDictionary?["name"] as! String?
+        self.getAssignees(assignees: stepDictionary?["assignees"] as! [String])
+        
+        if let details = stepDictionary?["details"] as? String?  {
+            self.details = details
+        }
+        if let state = stepDictionary?["state"] as? String?  {
+            self.state = state
+        }
+        if let completedAt = stepDictionary?["completed_at"] {
+            self.completedAt = (completedAt as! TimeInterval)
+        }
+        if let completedBy = stepDictionary?["completed_by"] {
+            getCompletedBy(completedBy: (completedBy as! String))
```

Comment:
It seems weird to me to a) do this async and b) make network requests from with the init function of a model. This might pose problems for us later especially when trying to display user information about who completed it or who signed off because the views will not know to update once the data has been fetched (and there's no good communication channel to make that happen)

We might want to consider doing all of this work within the Parse client. You can imagine an algorithm like:
* fetch the task
* fetch all of the users associated with the task (easy to derive)
* assign the user objects to the task where necessary (might be a bit involved)
* invoke the completion callback

This way, by the time the callback is invoked, we have all of the data needed for that task. What do you think?